File: Example/OCExample/Utility.m
//
//  Utility.m
//  OCExample
//
//  Created by xuyunshi on 2022/1/10.
//  Copyright © 2022 CocoaPods. All rights reserved.
//

#import "Utility.h"
#import <UIKit/UIKit.h>

Theme const ThemeAuto = @"auto";
Theme const ThemeDark = @"dark";
Theme const ThemeLight = @"light";

@implementation Utility

+ (UIButton *)buttonWith: (NSString *)title index: (int)index {
    UIButton* btn = [UIButton buttonWithType:UIButtonTypeCustom];
    btn.backgroundColor = [Utility randomColor];
    [btn setTitle:title forState:UIControlStateNormal];
    btn.tag = index;
    btn.contentEdgeInsets = UIEdgeInsetsMake(10, 10, 10, 10);
    return btn;
}

+ (UIColor *)randomColor {
    NSArray* colors = @[UIColor.systemRedColor,
                        UIColor.blackColor,
                        UIColor.systemOrangeColor,
                        UIColor.systemBlueColor,
                        UIColor.systemGreenColor,
                        UIColor.systemPinkColor,
                        UIColor.systemYellowColor,
                        UIColor.systemGrayColor,
                        UIColor.systemPurpleColor,
                        UIColor.systemTealColor];
    int i = random() % 10;
    return colors[i];
}

@end

File: Example/Fastboard/Utility.swift
//
//  Utility.swift
//  Fastboard_Example
//
//  Created by xuyunshi on 2022/1/7.
//  Copyright © 2022 CocoaPods. All rights reserved.
//

import UIKit

func button(title: String, index: Int) -> UIButton {
    let btn = UIButton(type: .custom)
    btn.backgroundColor = randomColor()
    btn.setTitle(title, for: .normal)
    btn.tag = index
    let t = UIApplication.shared.windows.first?.traitCollection
    let hasCompact = t?.verticalSizeClass == .compact || t?.horizontalSizeClass == .compact
    if hasCompact {
        btn.contentEdgeInsets = .init(top: 4, left: 10, bottom: 4, right: 10)
    } else {
        btn.contentEdgeInsets = .init(top: 10, left: 10, bottom: 10, right: 10)
    }
    return btn
}

func randomColor() -> UIColor {
    let indicates: [UIColor] = [
        .systemRed,
        .black,
        .systemOrange,
        .systemBlue,
        .systemGreen,
        .systemPink,
        .systemYellow,
        .systemGray,
        .systemPurple,
        .systemTeal
    ]
    let i = Int.random(in: 0..<10)
    return indicates[i]
}
